x-pack/plugins/profiling/server/routes/query.ts

```diff
@@ -40,24 +40,31 @@ export function createCommonFilter({
   };
 }
 
-export function autoHistogramSumCountOnGroupByField(searchField: string) {
+export function findFixedIntervalForBucketsPerTimeRange(
+  timeFrom: number,
+  timeTo: number,
+  buckets: number
+): string {
+  const range = timeTo - timeFrom;
+  const interval = Math.max(Math.floor(range / buckets), 1);
+  return `${interval}s`;
+}
+
+export function aggregateByFieldAndTimestamp(searchField: string, interval: string) {
   return {
-    auto_date_histogram: {
-      field: '@timestamp',
-      buckets: 50,
+    terms: {
+      field: searchField,
+      order: { count: 'desc' },
+      size: 100,
+      // 'execution_hint: map' skips the slow building of ordinals that we don't need.
+      // Especially with high cardinality fields, this setting speeds up the aggregation.
+      execution_hint: 'map',
```

Comment:
it might be a good idea to only use execution_hint: 'map' on known high cardinality fields. I think there's a pretty significant perf boost from using ordinals for low cardinality fields.

Reply:
Thanks, I didn't know about that performance boost. I confirmed that `StackTraceID` is the only high cardinality field for top N stacktraces. I refactored the code to enable `execution_hint: 'map'` only for high cardinality fields. Using the MVP cluster, I verified the boost for non-`StackTraceID` queries.